### PR TITLE
feature/build-pipeline-overhaul

### PR DESCRIPTION
**Pull request by busticated**

### CODE REVIEW

File: bin/testWiFi.js [hunk 1]
```diff
@@ -1,4 +0,0 @@
-var TestWiFi = require('../dist/lib/TestWifi');
-
-var test = new TestWiFi();
-test.run();
```
Comment (busticated) [inline bin/testWiFi.js]: i need to confirm this w/ the team

File: package.json [hunk 2]
```diff
@@ -124,32 +120,36 @@
...
+    "clean": "rm -rf ./dist",
+    "clean:modules": "rm -rf ./node_modules",
     "update-firmware-binaries": "node scripts/update-firmware-binaries.js",
-    "version": "npm run update-changelog",
-    "update-changelog": "VERSION=`node -p -e \"require('./package.json').version\"` bash -c 'read -p \"Update CHANGELOG.md for version $VERSION and press ENTER when done.\"' && git add CHANGELOG.md"
+    "update-changelog": "VERSION=`node --print --eval \"require('./package.json').version\"` bash -c 'read -p \"Update CHANGELOG.md for version $VERSION and press ENTER when done.\"' && git add CHANGELOG.md",
+    "reinstall": "npm run clean && npm run clean:modules && npm i",
+    "doctoc": "doctoc --title '## Table of Contents' README.md",
+    "todo": "git grep -e 'TODO (.*)' --ignore-case --heading --break --context=2 --full-name --line-number -- :^./package.json"
   },
```
Comment (busticated) [inline package.json]: I owe README updates

**Comment by monkbroc** [comment]
One thing I would do during acceptance testing or PR review is run `PARTICLE_CLI_DEVELOPMENT=1 bin/particle.js` to use the latest code without needing to do `npm run compile`. I noticed that I can get the same behavior with `node src/index.js` since babel-register is no longer necessary to transpile module import statements. I'll symlink to that file from `~/bin/particle.js` to allow me running the development version of the CLI.

I don't know if it's worth documenting in README.md but it's something I wanted to share.

**Comment by busticated** [comment]
@monkbroc 

> I'll symlink to that file from ~/bin/particle.js to allow me running the development version of the CLI

be aware that there _may_ still be situations where the transpiled code is needed (e.g. if your local node is `v6`)

we might also be able to wire something up leveraging `npm link`. currently if you run that within the repo, `particle` will link to the compiled version - e.g.

```
readlink $(which particle)

../lib/node_modules/particle-cli/dist/index.js
```

```
npm list -g --depth=0

/usr/local/lib
├── @particle/config@1.0.2 -> /Users/me/Sites/particle/config
├── npm@5.10.0
├── particle-cli@1.40.0 -> /Users/me/Sites/particle/cli
└── yarn@1.3.2
```

maybe we could add a `particle-dev` bin via:

https://github.com/particle-iot/particle-cli/blob/93ec27de31002ca75e0c8e94d80c10aa8bffbe43/package.json#L7

and somehow strip it out or disable it at publish-time.

**Comment by monkbroc** [comment]
> be aware that there may still be situations where the transpiled code is needed (e.g. if your local node is v6)

Understood. It's fine to document that running the non-transpiled code is only compatible with Node 10. We could keep ourselves honest by running the tests through Node 10 without Babel and expecting those to pass.

File: bin/testWiFi.js (hunk 1 above)
Comment (busticated) [reply bin/testWiFi.js]: all good 👍

File: package.json (hunk 2 above)
Comment (busticated) [reply package.json]: done 👉 https://github.com/particle-iot/particle-cli/pull/484/commits/0f5902b261b87aedd7cea24754dc996000560c92

**Pull request merged**